ballerina/tests/schema_client.bal

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents a mock database schema client.
+isolated client class MockSchemaClient {
+    *SchemaClient;
+
+    private final MockClient dbClient;
+    private final string database;
+
+    public function init(string url, string user, string password, string database) returns Error? {
+        self.database = database;
+        self.dbClient = check new(url, user, password);
+    }
+
+    isolated remote function listTables() returns string[]|Error {
```

Comment:
Why the implementation is added in the tests? This way we have to reimplement everything for other connectors isnt it?

Reply:
Hmm.. Lets proceed with this for this time. IMO it would be better if only queries are asked from each connectors. Other code should be in common module.